conda/activate.py

```diff
@@ -115,18 +115,30 @@ def get_export_unset_vars(self, export_metavars=True, **kwargs):
         for name, value in kwargs.items():
             if value is None:
                 unset_vars.append(name.upper())
+                if context.keep_case:
+                    unset_vars.append(name)
+
             else:
                 export_vars[name.upper()] = value
+                if context.keep_case:
+                    export_vars[name] = value
 
         if export_metavars:
             # split meta variables into exports vs unsets
             for name, value in context.conda_exe_vars_dict.items():
                 if value is None:
                     unset_vars.append(name.upper())
+                    if context.keep_case:
+                        unset_vars.append(name)
+
                 elif "/" in value or "\\" in value:
                     export_vars[name.upper()] = self.path_conversion(value)
+                    if context.keep_case:
+                        export_vars[name] = self.path_conversion(value)
```

Comment:
This clause is untested in `test_metavars_keep_case`. Let's adjust the mocked value for `context.conda_exe_vars_dict` to include a path-like value.